### PR TITLE
tests/server/config: migrate the tests to testify

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/4813.

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/server/config/config_test.go [hunk 1]
```diff
@@ -35,36 +35,36 @@ var dialClient = &http.Client{
...
-	s.cleanup()
-	s.cluster.Destroy()
+func (suite *configPersistTestSuite) TearDownSuite() {
+	suite.cleanup()
+	suite.cluster.Destroy()
 }
 
-func (s *testConfigPresistSuite) TestRateLimitConfigReload(c *C) {
-	leader := s.cluster.GetServer(s.cluster.GetLeader())
+func (suite *configPersistTestSuite) TestRateLimitConfigReload() {
```
Comment (JmPotato) [inline tests/server/config/config_test.go:64]: It seems that we only have one test for `configPersistTestSuite`, so there is no need to use suite here?

Comment (lhy1024) [reply tests/server/config/config_test.go:64]: I'm not sure if other tests will be added to this suite. cc @CabinfeverB

Comment (CabinfeverB) [reply tests/server/config/config_test.go:64]: There are no planned tests to be added

File: tests/server/config/config_test.go [hunk 2]
```diff
@@ -35,36 +35,19 @@ var dialClient = &http.Client{
...
-
-func (s *testConfigPresistSuite) TearDownSuite(c *C) {
-	s.cleanup()
-	s.cluster.Destroy()
-}
+	re.NoError(err)
+	re.NoError(cluster.RunInitialServers())
+	re.NotEmpty(cluster.WaitLeader())
+	defer cancel()
+	defer cluster.Destroy()
```
Comment (JmPotato) [inline tests/server/config/config_test.go:46]: Please put these defer functions to a closer place to where they are created.

File: tests/server/config/config_test.go [hunk 3]
```diff
@@ -73,26 +56,26 @@ func (s *testConfigPresistSuite) TestRateLimitConfigReload(c *C) {
...
-	c.Assert(err, IsNil)
+	re.NoError(err)
 	req, _ := http.NewRequest("POST", leader.GetAddr()+"/pd/api/v1/service-middleware/config", bytes.NewBuffer(data))
 	resp, err := dialClient.Do(req)
-	c.Assert(err, IsNil)
+	re.NoError(err)
 	resp.Body.Close()
-	c.Assert(leader.GetServer().GetServiceMiddlewarePersistOptions().IsRateLimitEnabled(), Equals, true)
-	c.Assert(leader.GetServer().GetServiceMiddlewarePersistOptions().GetRateLimitConfig().LimiterConfig, HasLen, 1)
+	re.Equal(true, leader.GetServer().GetServiceMiddlewarePersistOptions().IsRateLimitEnabled())
```
Comment (JmPotato) [inline tests/server/config/config_test.go:64]:
```suggestion
	re.True(leader.GetServer().GetServiceMiddlewarePersistOptions().IsRateLimitEnabled())
```

File: tests/server/config/config_test.go [hunk 4]
```diff
@@ -73,26 +56,26 @@ func (s *testConfigPresistSuite) TestRateLimitConfigReload(c *C) {
...
 	oldLeaderName := leader.GetServer().Name()
 	leader.GetServer().GetMember().ResignEtcdLeader(leader.GetServer().Context(), oldLeaderName, "")
-	mustWaitLeader(c, s.cluster.GetServers())
-	leader = s.cluster.GetServer(s.cluster.GetLeader())
+	mustWaitLeader(re, cluster.GetServers())
+	leader = cluster.GetServer(cluster.GetLeader())
 
-	c.Assert(leader.GetServer().GetServiceMiddlewarePersistOptions().IsRateLimitEnabled(), Equals, true)
-	c.Assert(leader.GetServer().GetServiceMiddlewarePersistOptions().GetRateLimitConfig().LimiterConfig, HasLen, 1)
+	re.Equal(true, leader.GetServer().GetServiceMiddlewarePersistOptions().IsRateLimitEnabled())
```
Comment (JmPotato) [inline tests/server/config/config_test.go:72]:
```suggestion
	re.True(leader.GetServer().GetServiceMiddlewarePersistOptions().IsRateLimitEnabled())
```

**Review by JmPotato — COMMENTED** [review]
The reset LGTM.

**Comment by lhy1024** [comment]
it is repeated

**Pull request closed without merging**